graph/graph.go

```diff
@@ -238,9 +363,22 @@ func (n *ArgumentNode) String() string {
 }
 
 func (g Graph) String() string {
-	nodeStrings := make([]string, len(g.Graph))
+	nodeStrings := make([]string, len(g.Edges)+len(g.ArgumentEdges)+2)
```

Comment:
set the capacity here instead of length, and use append instead of assigning by index.